chanzuckerberg/czgenepi
# Maya/192742 redux

### Summary
- **What:** Implement from end state management
  - include examples of how to write actions, selectors, and how to dispatch actions to update the store
- **Why:** Allow us to start storing certain state on the front end, such as current /last viewed group/pathogen 
- **Ticket:** [[sc-192742]](https://app.shortcut.com/genepi/story/192742)
- **Env:** https://maya-redux-frontend.dev.czgenepi.org/

### Testing
0. install [redux dev tools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) and restart chrome
1. load the app and make sure it doesn't crash
2. check out chrome dev tools `Application` tab and see that localstorage is empty
1. check out the Redux dev tools tab and see the state looks like this
>   ```
>     current: {
>       group: null,
>       pathogen: null,
>     },
>  ```

### Demos
![after](https://user-images.githubusercontent.com/7562933/176779447-750c476a-c53f-4d6c-8549-6a8160c5d8cf.gif)

### Next steps
- stop depending on /me to know what group is current
- if no group in localstorage, choosing to display the oldest group to the user by default
- updating api calls to inculde group id in path

### Checklist
